Creating the State Object

rockchalkwushock opened the pull request:
Please review index.js where I have build my state object and let me know if I'm on the right path. I wrote out some of the logic behind what should and maybe should go in the state object in the README.md file. Any good documentation, tutorials, or tidbits on this concept are always appreciated!

rockchalkwushock committed:
Initial Commit - contains all preliminary files with templates in place.

rockchalkwushock committed:
V1.1 Quiz-App-Refactored

added state_template.js
added Thinkful JS from GIST
added Thinkful HTML from GIST

rockchalkwushock committed:
V1.2 Creating the State

Began creating the state object in index.js. Commented areas code will
be needed in thinkful_code.js. Walked through my ideas of what should
be included in the state object in README.md

rockchalkwushock committed:
V1.3 Creating Actions

Still working on some of the state object.
Started making Action functions.

rockchalkwushock committed:
V1.4 Preparing to Debug

all prototypes are written a few more things to add before serious
debugging begins.

Holmberg18 commented on assets/js/state_template.js:
```diff
@@ -0,0 +1,22 @@
+// Event listeners
+$('.shopping-list-add').submit(function(event) {
+    event.preventDefault();
+    addItem(state, $('.shopping-list-add-input').val());
```
You can alternatively, invoke (e).preventDefault(); -- less typing, less prone to typos (same with "function(event)"

Holmberg18 commented on assets/js/temp2.js:
```diff
@@ -0,0 +1,18 @@
+    } else {
+      questions[index + 1].display = true;
+      // render function
+      index++;
```
I would suggest proper indentation for nested logic statements. Helps for better readability :)

Holmberg18 commented on assets/js/temp3.js:
```diff
@@ -0,0 +1,226 @@
+  questionHTML: null
+};
+
+var QUESTIONS = [
```
I would conform to lower case or camel case for object names, even with larger structures.

Holmberg18 commented on assets/js/temp3.js:
```diff
@@ -0,0 +1,226 @@
+            '2B',
+            'BAM128',
+            'Barely'
+        ],
```
I would personally put all answers in this array all on one line to be less prone to syntax errors.

Holmberg18 reviewed (commented):
Some minor tweaks to spacing, indentation and naming conventions.

rockchalkwushock committed:
V1.5

rockchalkwushock committed:
V1.6 Nearly there

Fixed getNextQuestions issue
Quiz Restarts but event listener is either not in place or not
responding need to look into.
Adding NAMESPACE.
Cleaning up code.
Trying to find area to implement `.bind()`

rockchalkwushock commented on README.md:
```diff
@@ -1,2 +1,24 @@
+What 'action' functions do I need to write for the program to run properly?
+
+  - must add question object to the state from `QUESTIONS[obj]`.
+  - must change score.
```
I was just taking what Thinkful gave me to work with, never done that before personally but I will change it in the final push.
